File: src/2399. Check Distances Between Same Letters.js
/**
 * @param {string} s
 * @param {number[]} distance
 * @return {boolean}
 */
function checkDistances(s, distance) {
  const pos = Array(26)
  for (let i = 0; i < s.length; i++) {
    const char = s[i],
      n = char.charCodeAt(0) - 97
    if (pos[n] > 0 && distance[n] !== i - pos[n]) return false
    pos[n] = i + 1
  }
  return true
}

File: src/2399. Check Distances Between Same Letters.cpp
class Solution {
public:
    bool checkDistances(string s, vector<int>& distance) {
        int pos[26] = {};
        for (int i = 0; i < s.size(); ++i) {
            int n = s[i] - 'a';
            if (pos[n] > 0 && distance[n] != i - pos[n]) return false;
            pos[n] = i + 1;
        }
        return true;
    }
};